File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution
{
public:
    int lengthOfLongestSubstring(string s)
    {
        unordered_map<char, int> mp;
        int subLen = 0;
        int start = 0;
        for (int i = 0; i < s.size(); i++)
        {
            while (mp[s[i]] && start < s.size())
            {
                mp[s[start]]--;
                start++;
            }
            mp[s[i]]++;
            subLen = max(subLen, i - start + 1);
        }

        return subLen;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int lengthOfLongestSubstring(String s) {
        int result = 0 ;
        int left=0;

        Map<Character,Integer> characterToIndexMap = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            Integer index = characterToIndexMap.get(s.charAt(i));
            if (index!=null&&index>=left && index<=i)
            {
                 left=index+1;
            }
            result=Math.max(result,i-left+1);
            characterToIndexMap.put(s.charAt(i),i);
        }
        return result;
    }
}